internal/interceptor.go

```diff
@@ -416,6 +421,11 @@ type ClientUpdateWorkflowInput struct {
 	WaitForStage        WorkflowUpdateStage
 }
 
+type ClientUpdateWithStartWorkflowInput struct {
+	UpdateInput            *ClientUpdateWorkflowInput
```

Comment:
This is input for another interceptor call, I don't think we should reuse it for this one. The interceptor input really just needs to be the values passed in by the user. Maybe something like:

```go
type ClientUpdateWithStartWorkflowInput struct {
    StartOptions  *StartWorkflowOptions
    StartWorkflowType string
    StartWorkflowArgs []any
    UpdateID string
    UpdateName string
    UpdateArgs []any
    UpdateWaitForStage WorkflowUpdateStage
}
```

Or something like:

```go
type ClientUpdateWithStartWorkflowInput struct {
    StartOptions  *StartWorkflowOptions
    StartWorkflowType string
    StartWorkflowArgs []any
    UpdateOptions *UpdateWorkflowOptions
}
```

Or something like:

```go
type ClientUpdateWithStartWorkflowInput struct {
    UpdateOptions *UpdateWorkflowOptions
    StartOperation WithStartWorkflowOperation
}
```

and add `Options() StartWorkflowOptions` and `WorkflowType() string` and `WorkflowArgs() []any` to `WithStartWorkflowOperation` interface.

Reply:
What problem are you trying to solve? This is how it's done in Java:

```java
    public WorkflowUpdateWithStartInput(
        WorkflowStartInput workflowStartInput, StartUpdateInput<R> workflowUpdateInput) {
```

Reply:
I'll note Java does reuse, but I think it is the only SDK that takes such an approach 

https://github.com/temporalio/sdk-java/blob/c7fcf12f4fc1965cde8a70feb0a109f198298088/temporal-sdk/src/main/java/io/temporal/common/interceptors/WorkflowClientCallsInterceptor.java#L223

Reply:
Ah right I see, thank you @Quinn-With-Two-Ns and @cretz for the suggested schemas. I've switched to 

```go
type ClientUpdateWithStartWorkflowInput struct {
	UpdateOptions          *UpdateWorkflowOptions
	StartWorkflowOperation WithStartWorkflowOperation
}
```

9562e0e4e64f6062aeb56fccde9362b610eaa913